tools/backup_analysis.py

```diff
@@ -0,0 +1,186 @@
+    print('  size in integer number of bytes of the incremental backup blocks.')
+    print('  The default is 16777216 (16MB).')
+    sys.exit(1)
+
+def die(reason):
+    print('backup_analysis.py: error: ' + reason, file=sys.stderr)
+    sys.exit(1)
+
+# Check that the directory given is a backup. Look for the WiredTiger.backup file.
+def check_backup(mydir):
+    if not os.path.isdir(mydir):
+        return False
+    backup_file = mydir + "/WiredTiger.backup"
+    if not os.path.exists(backup_file):
+        return False
+    return True
+
+def compare_file(dir1, dir2, filename, cmp_size, granularity):
+    f1_size = os.stat(os.path.join(dir1, filename)).st_size
+    f2_size = os.stat(os.path.join(dir2, filename)).st_size
+    min_size = min(f1_size, f2_size)
+    # Initialize all of our counters per file.
+    bytes_gran = 0
+    gran_blocks = 0
+    num_blocks = min_size // cmp_size
+    offset = 0
+    partial = min_size % cmp_size
+    pct20 = granularity // 5
+    pct80 = pct20 * 4
+    pct20_count = 0
+    pct80_count = 0
+    start_off = offset
+    total_bytes_diff = 0
+    fp1 = open(os.path.join(dir1, filename), "rb")
+    fp2 = open(os.path.join(dir2, filename), "rb")
+    # Time how long it takes to compare each file.
+    start = time.asctime()
+    # Compare the bytes in cmp_size blocks between both files.
+    print("")
+    for b in range(0, num_blocks + 1):
+        # Compare the two blocks. We know both files are at least min_size so all reads should work.
+        buf1 = fp1.read(cmp_size)
+        buf2 = fp2.read(cmp_size)
+        # If they're different, gather information.
+        if buf1 != buf2:
+            total_bytes_diff += cmp_size
+            # Count how many granularity level blocks changed.
+            if bytes_gran == 0:
+                gran_blocks += 1
+            bytes_gran += cmp_size
+        # Gather and report block information when we cross a granularity boundary or we're on
+        # the last iteration.
+        offset += cmp_size
+        if offset % granularity == 0 or b == num_blocks:
+            if bytes_gran != 0:
+                print(f'{filename}: offset {start_off}: {bytes_gran} bytes differ in {granularity} bytes')
+            # Account for small or large block changes.
+            if bytes_gran != 0:
+                if bytes_gran <= pct20:
+                    pct20_count += 1
+                elif bytes_gran >= pct80:
+                    pct80_count += 1
+            # Reset for the next granularity block.
+            start_off = offset
+            bytes_gran = 0
+
+    # Account for any partial blocks.
+    if partial != 0:
+        buf1 = fp1.read(partial)
+        buf2 = fp2.read(partial)
+        # If they're different, gather information.
+        if buf1 != buf2:
+            total_bytes_diff += partial
+            bytes_gran += partial
+            print(f'{filename}: offset {start_off}: {bytes_gran} bytes differ in {partial} bytes')
```

Comment:
I don't thin that `{partial}` is correct, as earlier this is the place where you would print 16MB (`granularity`), while `partial` is defined above as the last bytes that are less than 4KB. I think you should replace this with `offset + partial - start_off`?

Reply:
Fixed